### PR TITLE
[WIP] Vina prep

### PR DESCRIPTION
**Pull request by skearnes**
The PDBQTs look a bit ugly for some reason, even though the output from pdbfixer looks just fine. I'll keep working on this, but it's a good starting point (and may work just fine as is).

To get pdbfixer running, you'll need to install openmm (easiest from conda/binstar) and then download and install the v1.1 release of pdbfixer from github (the binstar version is broken).

### CODE REVIEW

File: vs_utils/scripts/public_data/vina_prep.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                      help='Input filename.')
+  parser.add_argument('-r', '--receptor', action='store_true',
+                      help='Treat input as a receptor.')
+  parser.add_argument('-s', '--split', action='store_true',
+                      help='Split output files into one file per molecule.')
+  return parser.parse_args(input_args)
+
+
+def main(input_filename, receptor=False, split=False):
+  output_filename = os.path.basename(input_filename)
```
Comment (rbharath) [inline vs_utils/scripts/public_data/vina_prep.py]: Would it be possible to factor these changes into a class? For my particular use case, I can't make use of subprocess when running python. If a class is tricky, a neat shell-script would actually be a good alternative as well.

Comment (skearnes) [inline vs_utils/scripts/public_data/vina_prep.py]: I think a shell script would be easiest, then. I have some previous commits here that will help with that.

**Comment by skearnes** [comment]
Added vina_prep.sh. Note that it won't play nicely with files that have dots in their names that are not for extensions (.sdf and .sdf.gz are fine; test.1.sdf.gz is not). PTAL.

File: vs_utils/scripts/public_data/vina_prep.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+if [[ ${split} = 'true' ]]
+then
+  opts="${opts} -m"
+fi
+
+output_basename=$(basename ${input_filename} | sed 's/\..*$//g')
+
+echo obabel ${input_filename} -O ${output_basename}.pdbqt ${opts}
+obabel ${input_filename} -O ${output_basename}.pdbqt ${opts}
```
Comment (rbharath) [inline vs_utils/scripts/public_data/vina_prep.sh]: Also, can you move in the pdbfixer commands here as well?

**Comment by rbharath** [comment]
Another request: Can we move a version of pdbfixer into vs_utils (like we did for rdkit-utils)? It's tough for me to get the install for a new library approved, but it's quite easy for me to do an in-place version upgrade for vs_utils.

**Comment by skearnes** [comment]
Do we have OpenMM available? It's a dependency for pdbfixer (I'm happy to move in the v1.1 release, but I want to make sure that helps us).

**Comment by rbharath** [comment]
OpenMM isn't currently available to us. I'm playing with making it
available, but I don't yet know how easy/hard that will be. Is there a way
we can avoid using OpenMM?

Do we need pdbfixer for ligands or only for receptors? If it's only for
receptors, we could make it so that only clean versions are available in
mol-net.

Bharath

**Comment by skearnes** [comment]
We should only need pdbfixer for the receptors. I like the idea of cleaning them offline since there will be relatively few of them. I'll still add the pdbfixer commands to the shell script, though, so we can use it offline.

**Comment by rbharath** [comment]
Ok, sgtm. Can you make two cleanup scripts then, one for receptors
(offline), and one for ligands (online)?

Bharath

**Comment by skearnes** [comment]
How do these look?

**Comment by rbharath** [comment]
Yes, these two look good. LGTM. Can you apply the pdb_prep to all the pdb files included in molecule-net? I'll run the ligand prep script online.

**Comment by skearnes** [comment]
OK, I made another change to add all missing atoms and now have PDBQTs for phase1 and DUD-E. Will merge soon if no other comments.

**Pull request merged**